varinfo/var_info.py

```diff
@@ -362,6 +362,39 @@ def get_required_dimensions(self, variables: Set[str]) -> Set[str]:
             if self.get_variable(dimension) is not None
         )
 
+    def get_missing_variable_attributes(self, variable_name: str) -> dict[str, Any]:
+        """Return a single set of all attributes for a variable that is not present
+        in the granule file(e.g. grid_mapping variable in collections that are not
+        fully CF compliant).
+
+        """
+        variable_attributes = self.cf_config.get_cf_attributes(variable_name)
+        return (
+            variable_attributes['cf_supplements'] | variable_attributes['cf_overrides']
+        )
+
+    def get_references_for_attribute(
+        self, list_of_variables: list[str], reference_attribute_name: str
+    ) -> set[str]:
+        """Return a single set of all references in a specific metadat attribute
+        for a list of variables (e.g. bounds, coordinates, cf attributes).
```

Comment:
Might want to refer back to `varinfo.utilities::CF_REFERENCE_ATTRIBUTES` to point people back to the full list of supported metadata attributes.

Reply:
add that reference to the comment?